### ISSUE TITLE
About the unsupervised loss

### ISSUE THREAD

**Issue by yassouali** [tarun005/USSS_ICCV19#1]
First of all, thank you for this wonderful  work and for providing the implementation

I am was interested in seeing how the unsupervised loss was implemented, in the paper, the loss is the entropy of sigmoid(v), the v is the similarity between the embedding of two labels set (either within or across the dataset).

I have two main questions if I may:
- In the implementation I see that we only calculate the entropy between the element products of embeddings directly, is the sigmoid applied before the loss calculation because I see that it was omitted in this line.
`loss_unsup = {d:beta*torch.mean(-1* torch.sum(log_probs * tensor_prob , dim=1)).view(-1)}`

- I see that in the intialization, we initialize the weights of the embedding layer using `torch.randn(NUM_LABELS , em_dim)`, does this help the training and it better than using the initiale weights of the conv layer in the beginning. And during training we don't update the weights of this conv1x1, can you please explain why is that.

Thank you.

**Comment by tarun005** [comment]
\sigma is not the sigmoid function, but the softmax function. Also, only the subroutine [_similarity](https://github.com/tarun005/USSS_ICCV19/blob/e8a21b54aeafa47bd96c55bf8cecf31fefacb25a/EntropyLoss.py#L101) is relevant to the paper, [_self_entropy](https://github.com/tarun005/USSS_ICCV19/blob/e8a21b54aeafa47bd96c55bf8cecf31fefacb25a/EntropyLoss.py#L159) is not.

**Comment by yassouali** [comment]
Thank you for the response, just one last thing :), is the conv1x1 (embedding -> n_labels) never being trained and only randomly initialized?

**Comment by tarun005** [comment]
The labels are initialized using the pretrained label embeddings [here](https://github.com/tarun005/USSS_ICCV19/blob/e8a21b54aeafa47bd96c55bf8cecf31fefacb25a/segment.py#L186). Yes, we do not train the label embeddings during the training process, as we have observed that it is not beneficial to do so (Section B.2 in the supplementary). It can be an interesting future work.

**Comment by yassouali** [comment]
Thank you!
